### ISSUE TITLE
About Xera's discontinuation & a call to the community

### ISSUE THREAD

**Issue by 4yx** [mahtab2003/Xera#49]
I saw that with the latest change, Xera was marked as discontinued despite the new changes that were made just before that.
I wanted to inquire as to the reason for that.

If the reason has to do with a breaking bug (perhaps #24, although it seems to be fixed), I understand there is not much to be done.
However, if Xera is retired because you're feeling overwhelmed with feature requests, and generally developing it (as indicated in [this forum post](https://fourm.xera.eu.org/d/63-callback-url-account-on-pending/11)), I'd ask you to reconsider.

Of course, there is no need to create everything you're being asked to. Saying something cannot be implemented now, or ever from you, while hard, is necessary so that time that could be devoted to other needs of the project is not wasted.
The same logic can be applied to creating new features of your own, or extending areas of the project. There is no need for everything to be implemented, or rushed. I believe a great majority would prefer a "slowly"-developed, but functional and secure, client area, than one with many features that overwhelm it's developer and force his hand in many situations.
I believe that if you were to continue maintaining Xera, even at a much slower pace than the one you did before, it would remain a very helpful client management system used by many in their navigation around MOFH.

With all that being said, there are of course other reasons why you could have chosen to stop maintaining it, like loss of interest, exploring/focusing on other areas, personal matters or something you wouldn't like to mention. It's understandable, and obviously creating a project does not mean you have any obligation to continue developing or supporting it. 

In this sense, my post is also a call to any community members that have the knowledge and relative time required to maintain a project like this. If the reason for Xera's discontinuation does not have to do with a non-fixable issue (for example, a bug with the MOFH API), do not let this project die. Please contribute, in any way you can, in keeping a community-led client area alive, either with forking the project or contributing your changes in another form (PRs perhaps, if @mahtab2003 is going to be checking them).
Needless to mention that the "no obligations" part applies here as well, as nobody is obligated to do so, or to continue from here and then have to keep going forever, this is merely my personal opinion/call to anyone interested in keeping the project alive.
Regarding the "the one asking" principle (basically, what is the one asking others to do something going to do himself), I have an extremely limited knowledge of PHP so I obviously cannot assist with keeping the project alive, but I can offer my assistance in other areas (documentation writing, translating etc.) if it is needed by those who can and want to do so.

In any case, as this has been a very long post, I'm essentially asking for the reason the project was discontinued, and asking any willing members of the community to keep the project alive.

@mahtab2003, thank you for all the work you've put in creating, developing and maintaining Xera for all this time, and thanks to all contributors who've helped in the journey!

**Comment by greenreader9** [comment]
I would be happy to manage this project after I have a bit more free time.

Many of the people here know of my affiliation with MOFH, and some of you are even beta testers for me free host.

After I have a bit more time and some workload goes down, I would be more then happy to manage this awesome project, and perhaps add in some code of my own for more features.

**Comment by 4yx** [comment]
> I would be happy to manage this project after I have a bit more free time.
> 
> Many of the people here know of my affiliation with MOFH, and some of you are even beta testers for me free host.
> 
> After I have a bit more time and some workload goes down, I would be more then happy to manage this awesome project, and perhaps add in some code of my own for more features.

Awesome! As one of your testers I can confirm your work is very good, and I'd love to see you extend Xera's features with some of your own.

**Comment by AgentDeath253** [comment]
i would help after i learn php, CSS, HTML

**Comment by 4yx** [comment]
> i would help after i learn php, CSS, HTML

A noble goal, good luck! I'd set a similar one for myself, but then I know I'd fail to honor it.

**Comment by gilect-com** [comment]
Xera's development has been going smoothly so far, no major bugs or security flaws (knock on wood).

The error with the Callback URL (Issue #24) that you mentioned has never occurred for me or other people (even using https). There isn't enough information to replicate this issue for it to be thoroughly tested.

Supposedly, the "issue" has been fixed now, though. Maybe there was something wrong with the code for that specific version of Xera, and not the previous ones, I don't know.

Right now, I think the biggest hurdle in development would be upgrading to CodeIgniter 4, as mentioned by a user of the Xera forum:
https://fourm.xera.eu.org/d/67-feature-request-some-great-ideas/18

I also think another problem is that people are asking for many features that may be difficult to implement. Let's Encrypt SSL, Self Signed Certificates, multiple admins, etc, without taking time to consider the implications of such suggestions.

The focus should be more on refining Xera, polishing it, and having a really good client area and client management panel. Rather, the focus currently seems to be on cramming Xera with as many features as possible.

A lot of the work done by @mahtab2003 seems to go underappreciated. So, as @4yx stated:

> @mahtab2003, thank you for all the work you've put in creating, developing and maintaining Xera for all this time, and thanks to all contributors who've helped in the journey!

**Comment by 4yx** [comment]
> Right now, I think the biggest hurdle in development would be upgrading to CodeIgniter 4, as mentioned by a user of the Xera forum: https://fourm.xera.eu.org/d/67-feature-request-some-great-ideas/18

Wow, and to think that all this time I thought it was written on CodeIgniter 4. Shows how much I know, huh?
Given that, if I remember and had read correctly, 4 was a major rewrite, I can see how that would be a very time-consuming (and of course, not easy) task. Perhaps projects such as [ci3-to-4-upgrade-helper](https://github.com/kenjis/ci3-to-4-upgrade-helper) could help, although the project says it's still under early development.

> I also think another problem is that people are asking for many features that may be difficult to implement. Let's Encrypt SSL, Self Signed Certificates, multiple admins, etc, without taking time to consider the implications of such suggestions.

I agree, and while I believe that users should be able to express their requests freely, I understand how it feels from the developer's perspective, you can easily get overwhelmed trying to implement everything that's being asked. 

I think making a roadmap (unless there is a private one I don't know of, of course) and postponing them for a very different time would be crucial (finding out which tasks need to be completely in a more urgent matter, and which can wait for a long time period), however I'm one to talk, if you find any roadmap in my projects be sure to let me know 😂. 
Generally, anyone that devotes their time to Xera should ensure they do not get stressed by attempting to fulfill everyone's requests; the "PRs welcome" attitude of FOSSBilling comes to mind, and alternatively letting users know it's considered but unknown when or if it will be implemented is also a good strategy.

> The focus should be more on refining Xera, polishing it, and having a really good client area and client management panel. Rather, the focus currently seems to be on cramming Xera with as many features as possible.

Absolutely agree with this. Having many features is great; having many features at the expense of a refined and polished project, the developers' physical and mental health and in the end, the users' experience, is a project-killer.

**Comment by AgentDeath253** [comment]
i could make a more detailed docs for now

**Comment by mahtab2003** [comment]
I will still accept pull request. But i will not contribute in Project Xera because it takes much time and focus so I'm unable to focus on my studies. Xera is too too too much complex.

**Comment by mahtab2003** [comment]
I will introduce something new after a few months.

**Comment by 4yx** [comment]
> I will still accept pull request. But i will not contribute in Project Xera because it takes much time and focus so I'm unable to focus on my studies. Xera is too too too much complex.

Ah, I see. Good luck with your studies, take as much time as needed! Xera is very time consuming indeed.

> I will introduce something new after a few months.

Glad to hear you still have plans for it, prioritizing your studies is important though so don't feel rushed to make it. I'm sure it can wait like you mentioned.

Once again, thanks for everything you've made so far and plan to keep on doing, and good luck with your studies!

**Comment by tanguy-fgl** [comment]
Maybe you can add some permissions to a person to publish release and accept pulls

**Comment by AgentDeath253** [comment]
> Maybe you can add some permissions to a person to publish release and accept pulls

that would be smart

**Comment by 4yx** [comment]
> Maybe you can add some permissions to a person to publish release and accept pulls

I agree with that as well. A new release, considering all the changes in https://github.com/mahtab2003/Xera/compare/1.2.8...dev would be nice, and having another person to review and accept pull requests would make the process easier.

**Comment by tanguy-fgl** [comment]
If you want I can do the job

**Comment by roshan1885** [comment]
I would be slowly contributing to this project as much as I can. Since this is a much great project which changed my whole system. As per as my studies, I will try to contribute, but not a much since self study is kinda difficult so i need to focus. 
I would suggest one more think that Mahtab should give one active person permissions to close issues and accept pull requests.

**Comment by chippytech** [comment]
I forked Xera and made PROJECT REMOVED

**Comment by 4yx** [comment]
> I forked Xera and made [GenerateArea](https://github.com/GenerateApps/GenerateArea).

Best of luck!

**Comment by mahtab2003** [comment]
Xera development is live again!

**Comment by santydesignscr** [comment]
I will start collaborating in this project, currently I worked with some features and fixing some bugs (congratulations to all contributers, there are too few).
